src/main/java/org/tarantool/MsgPackLite.java

```diff
@@ -91,6 +95,25 @@ public void pack(Object item, OutputStream os) throws IOException {
                 out.write(MP_DOUBLE);
                 out.writeDouble((Double) item);
             } else {
+                if (item instanceof BigInteger) {
+                    BigInteger value = (BigInteger) item;
+                    boolean isPositive = value.signum() >= 0;
+                    if (isPositive && value.compareTo(BI_MAX_64BIT) > 0 ||
+                            value.compareTo(BI_MIN_LONG) < 0)
+                        throw new IllegalArgumentException(
+                            "Cannot encode BigInteger as MsgPack: out of -2^63..2^64-1 range");
+                    if (isPositive && value.compareTo(BI_MAX_LONG) > 0) {
+                        byte[] data = value.toByteArray();
+                        // data can contain leading zero bytes
+                        int offset = 0;
+                        while (offset < data.length && data[offset] == 0)
+                            ++offset;
```

Comment:
I've found that the assert and the preceding while-loop are redundant in the if-block. 
A condition of execution of the if-block is (at least): value.compareTo(BI_MAX_64BIT) <= 0 and value.compareTo(BI_MAX_LONG) > 0. Which guarantees that all bytes in the data[] will be zero starting from 0 position to {data.length - 9} position inclusively (in case when data.size() > 8).

Reply:
To be honest, I'm feel uncomfortable w/o actual checking that leading bytes are zero. But I had rewrote the code to split out loop with asserts from the main logic. Hope it'll look more clean.